src/browser/fixture.js

```diff
@@ -31,19 +31,41 @@ function getComposedChildren(node) {
 
 async function waitForElem(elem, awaitLoadingComplete = true) {
 
-	const update = elem?.updateComplete;
-	if (typeof update === 'object' && Promise.resolve(update) === update) {
-		await update;
-		await nextFrame();
-	}
+	const doWait = async() => {
 
-	if (awaitLoadingComplete && typeof elem?.getLoadingComplete === 'function') {
-		await elem.getLoadingComplete();
-		await nextFrame();
-	}
+		const update = elem?.updateComplete;
+		if (typeof update === 'object' && Promise.resolve(update) === update) {
+			await update;
+			await nextFrame();
+		}
+
+		if (awaitLoadingComplete && typeof elem?.getLoadingComplete === 'function') {
+			await elem.getLoadingComplete();
+			await nextFrame();
+		}
 
-	const children = getComposedChildren(elem);
-	await Promise.all(children.map(e => waitForElem(e, awaitLoadingComplete)));
+		const children = getComposedChildren(elem);
+		await Promise.all(children.map(e => waitForElem(e, awaitLoadingComplete)));
+
+	};
+
+	await new Promise((resolve) => {
+		const observer = new MutationObserver((records) => {
+			for (const record of records) {
+				for (const removedNode of record.removedNodes) {
+					if (removedNode === elem) {
+						observer.disconnect();
+						resolve();
+						return;
+					}
+				}
+			}
+		});
+		observer.observe(elem.parentNode, { childList: true });
```

Comment:
This is the only part I don't love. Each child observes its parent looking for its own removal. I couldn't come up with a good way to do the more optimal alternative since we're not guaranteed that everything in `getComposedChildren()` will also be observable using a `MutationObserver`.

Reply:
What things aren't observable using a `MutationObserver`?

Reply:
It's just that `getComposedChildren` gets a `<slot>`'s assigned nodes and also traverses into a node's `shadowRoot`. `MutationObserver`'s `childNodes` isn't going to be a 1:1 match.

Reply:
Gotcha, and I assume you can't observe yourself